src/test/java/StepDefinition/DashboardTestSteps.java

```diff
@@ -0,0 +1,69 @@
+package StepDefinition;
+
+import com.ui.DashboardPage;
+import com.ui.LoginReportPortalPage;
+import com.ui.WidgetPage;
+import cucumber.TestContext;
+
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.testng.Assert;
+import utilities.ReadPropertyFile;
+
+import java.io.IOException;
+
+public class DashboardTestSteps {
+
+    TestContext testContext;
+    LoginReportPortalPage loginReportPortalPage;
+    DashboardPage dashboardPage;
+    WidgetPage widgetPage;
+    public ReadPropertyFile fr = new ReadPropertyFile();
+
+    public DashboardTestSteps(TestContext context) throws IOException {
+        testContext = context;
+        loginReportPortalPage = testContext.getPageObjectManager().getLoginReportPortalPage();
+        dashboardPage = testContext.getPageObjectManager().getDashboardPage();
+        widgetPage = testContext.getPageObjectManager().getWidgetPage();
+    }
+
+    @Given("User login the report portal")
+    public void userLogin() throws Exception {
+        loginReportPortalPage.loginToReportPortal(fr.getPropertyValue("reportPortal"), fr.getPropertyValue("username1"), fr.getPropertyValue("password1"));
+    }
+
+    @When("User creates the dashboard with  given dashboard name {string} and description {string}")
+    public void userEntersTheDashboardDetails(String dashname, String dashdesc) throws Exception {
+        dashboardPage.addDashboard(dashname, dashdesc);
+    }
+
+    @Then("Verify the dashboard {string} added successfully")
+    public void validateTheDashboardAdded( String dashName) {
+        Assert.assertTrue(dashboardPage.getDashboardTitle().equalsIgnoreCase(dashName));
+
+    }
+
+    @When("User adds the widget to the dashboard with {string} {string} {string} {string}")
```

Comment:
Please, add the names to the {strings} as it ambiguous what they are meaning. like:
User adds the widget with name: {widgetName}, description: {widgetDesc} and filter: {widgetFilter} to the {dashboardName} dashboard.
It will be more clear.